src/trees/SerialFunctionTree.cpp

```diff
@@ -67,8 +67,9 @@ SerialFunctionTree<D>::SerialFunctionTree(FunctionTree<D> *tree, SharedMemory *m
         this->maxNodesPerChunk = (sizePerChunk / this->sizeNodeCoeff / sizeof(double) / 8) * 8;
     }
 
-    this->lastNode = (ProjectedNode<D> *)this->sNodes; // position of last allocated node
-    this->lastGenNode = this->sGenNodes;               // position of last allocated Gen node
+    this->lastNode =
+        (ProjectedNode<D> *)this->sNodes; // position just after last allocated node, i.e. where to put next node
```

Comment:
`static_cast`. Can you also move the comment to the line above to avoid line break from clang-format?